walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -270,5 +274,156 @@ public void notEnoughRoom() {
 
     Assert.assertEquals(expected, actual);
   }
-}
 
+  @Test
+  public void optionalNotAttending() {
```

Comment:
There are many more behaviors you may want to test:

#1 special inputs
   * events is empty
   * request is empty
   * request wants to take > 24 hours
   * illegal inputs, e.g. illegal events, negative start and end, etc.
   * .... please think all the possible inputs. If other code doesn't make promises, do not go with your own assumptions.
#2 behavior for finding time slots
   * No optional attendee
   * No mandatory attendee
   * Request doesn't contain attendee
   * No timeslot for the meeting
   * ... same with above, try to think about all the possible cases and make sure all of the execution paths of your code could be covered by tests.

Reply:
Thank you for the detailed comment!

There already were 3 tests with NO_EVENTS written in the `FindMeetingQueryTest.java` file, so I hadn't added one. Same for NO_ATTENDEES, no timeslot for the meeting & no optional attendee. I will add tests for the special inputs that I haven't tested 👍🙂

Reply:
Could you document what you have tested in your doc to make sure you have full test coverage here?